py-rattler/rattler/prefix/prefix_record.py

```diff
@@ -3,25 +3,19 @@
 from typing import List, Optional
 from rattler.package.package_name import PackageName
 
-from rattler.rattler import PyPrefixRecord
+from rattler.rattler import PyRecord
 from rattler.prefix.prefix_paths import PrefixPaths
-from rattler.version.with_source import VersionWithSource
+from rattler.repo_data.record import RepoDataRecord
 
 
-class PrefixRecord:
-    def __init__(self, source: str) -> None:
-        if not isinstance(source, str):
-            raise TypeError(
-                "PrefixRecord constructor received unsupported type "
-                f" {type(source).__name__!r} for the `source` parameter"
-            )
-        self._record = PyPrefixRecord(source)
+class PrefixRecord(RepoDataRecord):
+    _record: PyRecord
```

Comment:
Does `PackageRecord` also specify this? Since the type is always `PyRecord` there is no need to duplicate that here (or in `RepoDataRecord`).